nexus-repository-helm/src/main/java/org/sonatype/repository/helm/HelmUploadHandler.java

```diff
@@ -94,11 +115,11 @@ public UploadResponse handle(Repository repository, ComponentUpload upload) thro
         throw new ValidationErrorsException("Metadata is missing the version attribute");
       }
 
-      String path = name + "-" + version + "." + TGZ_EXTENSION;
+      String path = String.format("%s-%s%s", name, version, extension);
 
       ensurePermitted(repository.getName(), HelmFormat.NAME, path, Collections.emptyMap());
       return TransactionalStoreBlob.operation.withDb(storageFacet.txSupplier()).throwing(IOException.class)
```

Comment:
Transaction should be opened in `upload()` method

Reply:
Yeap, it is a good point, but method upload is called in 2 two places. 
First place has already opened transaction, second has not opened transaction. So if there will be an added transaction start, this method must have to decide to open a transaction or use the current one, also current transaction implementation does not have a way to check if a transaction exists. By the way, implementation has UnitOfWork.currentTx() and if you will look at implementation you will see that this one can throw an exception if a transaction has not opened.
I think that this implementation is better for us because, in all formats, the transaction opens in UploadHandler. 
But was a little refactoring, please look at my changes 
https://github.com/sonatype-nexus-community/nexus-repository-helm/pull/84/files#diff-716689b8f314cb356bf1fe42dcbdea24R120-R124